### PR TITLE
`compute` - Fix acc tests

### PR DESCRIPTION
**Pull request by myc2h6o**
Fixing:
TestAccDedicatedHost_basicNewSku
TestAccLinuxVirtualMachineScaleSet_imagesPlan
TestAccLinuxVirtualMachineScaleSet_otherEdgeZone
TestAccWindowsVirtualMachine_otherEdgeZone
TestAccWindowsVirtualMachineScaleSet_otherEdgeZone
TestAccWindowsVirtualMachineScaleSet_extensionAutomaticUpgradeEnabled

### CODE REVIEW

File: internal/services/compute/dedicated_host_resource_test.go [hunk 1]
```diff
@@ -202,7 +202,7 @@ resource "azurerm_dedicated_host" "test" {
   name                    = "acctest-DH-%d"
   location                = azurerm_resource_group.test.location
   dedicated_host_group_id = azurerm_dedicated_host_group.test.id
-  sku_name                = "DCSv2-Type1"
+  sku_name                = "FSv2-Type2"
```
Comment (myc2h6o) [inline internal/services/compute/dedicated_host_resource_test.go:205]: previous type is not available in all regions in the acc test

File: internal/services/compute/linux_virtual_machine_scale_set_images_resource_test.go [hunk 2]
```diff
@@ -688,9 +688,9 @@ func (r LinuxVirtualMachineScaleSetResource) imagesPlan(data acceptance.TestData
 %s
 
 resource "azurerm_marketplace_agreement" "test" {
-  publisher = "cloudbees"
-  offer     = "jenkins-operations-center"
-  plan      = "jenkins-operations-center-solo"
+  publisher = "cloudwhizsolutions"
+  offer     = "jenkins-docker-container-with-ubuntu-server"
+  plan      = "jenkins-docker-container-with-ubuntu-server-cw"
```
Comment (myc2h6o) [inline internal/services/compute/linux_virtual_machine_scale_set_images_resource_test.go:693]: previous image is not available any more

File: internal/services/compute/linux_virtual_machine_scale_set_other_resource_test.go [hunk 3]
```diff
@@ -1071,7 +1071,7 @@ resource "azurerm_linux_virtual_machine_scale_set" "test" {
   name                = "acctestvmss-%d"
   resource_group_name = azurerm_resource_group.test.name
   location            = azurerm_resource_group.test.location
-  sku                 = "Standard_F2"
+  sku                 = "Standard_D2s_v3" # intentional for premium/edgezones
```
Comment (myc2h6o) [inline internal/services/compute/linux_virtual_machine_scale_set_other_resource_test.go:1074]: Apply same fix as https://github.com/hashicorp/terraform-provider-azurerm/blob/38d7ce51a7956716eca0db5a730603938eb83768/internal/services/compute/linux_virtual_machine_resource_other_test.go#L1235

File: internal/services/compute/windows_virtual_machine_scale_set_extensions_test.go [hunk 4]
```diff
@@ -1129,8 +1129,8 @@ resource "azurerm_windows_virtual_machine_scale_set" "test" {
   extension {
     name                       = "AzurePolicyforWindows"
     publisher                  = "Microsoft.GuestConfiguration"
-    type                       = "ConfigurationforLinux"
-    type_handler_version       = "1.0"
+    type                       = "ConfigurationforWindows"
+    type_handler_version       = "1.29"
```
Comment (myc2h6o) [inline internal/services/compute/windows_virtual_machine_scale_set_extensions_test.go:1133]: This is windows vmss, needs to use `ConfigurationforWindows`, and type 1.0 is not available any more

**Comment by myc2h6o** [comment]
Test result
image

**Review by katbyte — APPROVED** [review]
LGTM 🧮

**Pull request merged**